protocol/transport.go

```diff
@@ -3,33 +3,39 @@ package protocol
 import (
 	"encoding/binary"
 	"fmt"
+	"github.com/jackc/pgx/pgproto3"
 	"io"
 )
 
 // NewTransport creates a protocol
 func NewTransport(r io.Reader, w io.Writer) *Transport {
+	backend, _ := pgproto3.NewBackend(r, nil)
 	return &Transport{
-		R: r,
-		W: w,
+		R:       r,
+		W:       w,
+		backend: backend,
```

Comment:
why do we need to keep both R and backend (wrapper for R)? is it only for startup?
if so, what do you think about one of the following:
- create transport only after startup was done
- use backend.ReceiveStartupMessage